libs/data-types/src/lib/utilities/bootstrap-dynamic-types/bootstrap-dynamic-types.ts

```diff
@@ -0,0 +1,202 @@
+                acc.set(discriminantValue, ctor),
+            new Map<string | number, Ctor<Object>>()
+        );
+
+        unionMap.set(nameOfUnionToValidate, membersMapForThisUnion);
+    });
+
+    return unionMap;
+};
+
+const mixUnionMemberSchemasIntoTypeDefinitionForClass = (
+    allCtorCandidates: unknown[],
+    dataSchema: ReturnType<typeof getCoscradDataSchema>
+): CoscradPropertyTypeDefinition => {
+    const updatedSchema = Object.entries(dataSchema).reduce(
+        (acc, [propertyKey, propertyTypeDefinition]) => {
+            if (
+                isSimpleCoscradPropertyTypeDefinition(propertyTypeDefinition) ||
+                !['NESTED_TYPE', 'UNION_TYPE'].includes(propertyTypeDefinition.complexDataType)
+            )
+                return {
+                    ...acc,
+                    [propertyKey]: propertyTypeDefinition,
+                };
+
+            const { complexDataType } = propertyTypeDefinition;
+
+            if (complexDataType === 'NESTED_TYPE') {
+                const { schema: nestedSchema } = propertyTypeDefinition;
+
+                return {
+                    ...acc,
+                    [propertyKey]: {
+                        ...propertyTypeDefinition,
+                        schema: mixUnionMemberSchemasIntoTypeDefinitionForClass(
+                            allCtorCandidates,
+                            // @ts-expect-error fix me
+                            nestedSchema
+                        ),
+                    },
+                };
+            }
+
+            if (complexDataType === 'UNION_TYPE') {
+                const { unionName } = propertyTypeDefinition;
+
+                return {
+                    ...acc,
+                    [propertyKey]: {
+                        ...propertyTypeDefinition,
+                        schemaDefinitions: resolveMemberSchemasForUnion(
+                            allCtorCandidates,
+                            unionName
+                        ),
+                    },
+                };
+            }
+        },
+        {}
+    );
+
+    return updatedSchema as CoscradPropertyTypeDefinition;
+};
+
+export const bootstrapDynamicTypes = (allCtorCandidates: unknown[]) => {
+    /**
+     * Now we iterate through all classes with `CoscradDataType` definitions,
+     * writing the schemas for each union property.
+     */
+    const ctorsWithTypeDefinitions = allCtorCandidates.filter(
+        (input) => !isNullOrUndefined(getCoscradDataSchema(input))
+    );
+
+    const ctorsThatLeverageAUnionType = ctorsWithTypeDefinitions.filter((ctor) =>
+        leveragesUniontype(getCoscradDataSchema(ctor))
+    );
+
+    ctorsThatLeverageAUnionType.forEach((ctor) => {
+        const updatedSchema = mixUnionMemberSchemasIntoTypeDefinitionForClass(
+            allCtorCandidates,
+            getCoscradDataSchema(ctor)
+        );
+
+        // @ts-expect-error fix types
```

Comment:
Fix this. It's best to have an abstraction around writing this metadata and some type safety to deal with the recurring, annoying issue that property type decorators take the class constructor's prototype as the target.

Reply:
At a higher level, we just want to deal with the class ctor references directly.